File: src/main/java/org/visual/factory/GuiceFactory.java
package org.visual.factory;

import com.google.inject.ConfigurationException;
import org.visual.context.DIContext;
import picocli.CommandLine;

public class GuiceFactory implements CommandLine.IFactory {
  @Override
  public <K> K create(Class<K> aClass) throws Exception {
    try {
      return DIContext.INSTANCE.get(aClass);
    } catch (ConfigurationException ex) { // no implementation found in Guice configuration
      return CommandLine.defaultFactory().create(aClass); // fallback if missing
    }
  }
}

File: executable/visual-debugger/src/main/kotlin/org/visual/debugger/factory/GuiceFactory.kt
package org.visual.debugger.factory

import com.google.inject.ConfigurationException
import org.visual.debugger.context.DebuggerContext
import picocli.CommandLine

class GuiceFactory : CommandLine.IFactory {
    override fun <K : Any?> create(p0: Class<K>): K {
        return try {
            DebuggerContext.get(p0)
        } catch (ex: ConfigurationException) {
            CommandLine.defaultFactory().create(p0)
        }
    }
}